### PR TITLE
feat: migrate `/user/{user}/comments` to React

### PR DESCRIPTION
**Pull request by wescopeland**
This PR builds on https://github.com/RetroAchievements/RAWeb/pull/2762 by migrating the user comments page to React. There is actually very little new UI code here. A large chunk of this PR is just code coverage, autogenerated code, and refactoring to make stuff more reusable.

Screenshot 2024-10-24 at 10 02 38 PM

Behaviors I've verified:
* If the user's wall is disabled, other users cannot visit it.
* Policies are still respected.

### CODE REVIEW

File: app/Community/Controllers/UserCommentController.php [hunk 1]
```diff
@@ -4,29 +4,49 @@
...
+            routeName: 'user.comment.index',
+            routeParam: 'user',
+            view: 'user/[user]/comments',
+            createPropsData: function ($user, $paginatedComments, $isSubscribed, $me) {
+                return new UserCommentsPagePropsData(
+                    targetUser: UserData::fromUser($user)->include('id'),
+                    paginatedComments: PaginatedData::fromLengthAwarePaginator(
+                        $paginatedComments,
+                        total: $paginatedComments->total(),
+                        items: CommentData::fromCollection($paginatedComments->getCollection())
```
Comment (Jamiras) [inline app/Community/Controllers/UserCommentController.php:43]: I get the following error accessing any user's comments list.
```
Attempted to lazy load [commentable] on model [App\Models\UserComment] but lazy loading is disabled.
```
It seems to be coming from this line.

Comment (wescopeland) [reply app/Community/Controllers/UserCommentController.php:43]: Looks like it was an issue in a policy. Fixed in latest.

File: app/Policies/UserCommentPolicy.php
Comment (Jamiras) [inline app/Policies/UserCommentPolicy.php:1]: > App\Policies\UserCommentPolicy::create(): Argument #1 ($user) must be of type App\Models\User, null given, called in /var/www/html/app/Community/Controllers/UserCommentController.php on line 46

When trying to view comments while not logged in.

Comment (wescopeland) [reply app/Policies/UserCommentPolicy.php:1]: Fixed in latest and created an additional test case.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**